src/github_integration/github_manager.py

```diff
@@ -0,0 +1,231 @@
+        If a 'since' datetime is provided, fetches all pull requests since that time.
+        If a 'state' is provided, fetches pull requests with that state.
+
+        :param since: The datetime to fetch pull requests since. If None, fetches all pull requests.
+        :param state: The state of the pull requests to fetch. If None, fetches pull requests of all states.
+        :return: A list of PullRequest objects.
+        """
+        logging.info(f"Fetching all closed PRs for {self.__repository.full_name}")
+        if state is None:
+            pulls = self.__repository.get_pulls()
+        else:
+            pulls = self.__repository.get_pulls(state=state)
+
+        pull_requests = []
+        logging.info(f"Found {len(list(pulls))} PRs for {self.__repository.full_name}")
+        for pull in list(pulls):
+            pull_requests.append(PullRequest(pull))
+
+        return pull_requests
+
+    def fetch_commits(self, since: datetime = None) -> list[Commit]:
+        logging.info(f"Fetching all commits {self.__repository.full_name}")
+        raw_commits = self.__repository.get_commits()
+
+        commits = []
+        for raw_commit in raw_commits:
+            # for reference commit author - use raw_commit.author
+
+            # logging.debug(f"Raw Commit: {raw_commit}, Author: {raw_commit.author}, Commiter: {raw_commit.committer}.")
+            # logging.debug(f"Raw Commit.commit: Message: {raw_commit.commit.message}, Author: {raw_commit.commit.author}, Commiter: {raw_commit.commit.committer}.")
+
+            commits.append(Commit(raw_commit))
+
+        return commits
+
+    # get methods
+
+    def get_change_url(self, tag_name: str) -> str:
+        if self.__git_release:
+            # If there is a latest release, create a URL pointing to commits since the latest release
+            changelog_url = f"https://github.com/{self.__repository.full_name}/compare/{self.__git_release.tag_name}...{tag_name}"
+
+        else:
+            # If there is no latest release, create a URL pointing to all commits
+            changelog_url = f"https://github.com/{self.__repository.full_name}/commits/{tag_name}"
+
+        return changelog_url
+
+    def get_repository_full_name(self) -> Optional[str]:
+        """
+        Gets the full name of the repository.
+
+        :return: The full name of the repository as a string, or None if the repository is not set.
+        """
+        if self.__repository is not None:
+            return self.__repository.full_name
+        return None
+
+    # others
+
+    def show_rate_limit(self):
+        if not logging.getLogger().isEnabledFor(logging.DEBUG):
+            # save API Call when not in debug mode
+            return
+
+        if self.__g is None:
+            logging.error("GitHub object is not set.")
+            return
+
+        rate_limit: RateLimit = self.__g.get_rate_limit()
+
+        if rate_limit.core.remaining < rate_limit.core.limit/10:
```

Comment:
these `10` constants could be extracted to some common place

Reply:
Addressed in commit a403ed63ba1f547eccee98fdd739e0b8fc7e6255.